src/lib/y2network/interface.rb

```diff
@@ -31,10 +31,9 @@ def initialize(name)
 
     # Determines whether two interfaces are equal
     #
-    # @param other [Interface,:any] Interface to compare with
+    # @param other [Interface] Interface to compare with
     # @return [Boolean]
     def ==(other)
-      return false if other == :any
       name == other.name
```

Comment:
`nil.name` crash?

Reply:
maybe have something like `return false unless other.is_a?(Interface)` ?

Reply:
An interface is expected, we can return if classes differs of in case of other absence, I just unified with the rest.

Reply:
this won't work. Here it will crash or modify Route to not do this comparison for interface if it is nil https://github.com/yast/yast-network/pull/791/files/66ed1669abc644fee168701832c3701310c5dc09#diff-6901639be33a5a1baf300f012c547db3R54

Reply:
True, so, adding explicit return.